### ISSUE TITLE
Pretrained model: SG, GS messed up?

### ISSUE THREAD

**Issue by mishast** [leftthomas/CGD#10]
Strange.
I have similar images 's1.jpg', 's2.jpg'

**Case 1:**

model_state_file = 'sop_uncropped_resnet50_SG_1536_0.1_0.5_0.1_128_model.pth'
gd_config = 'SG'
num_classes = 11318
state_dict = torch.load(model_state_file, map_location=torch.device('cpu'))
net = Model('resnet50', gd_config, 1536, num_classes)
net.load_state_dict(state_dict)
net.eval()

Loss (Equlidean) = 0.48

**Case 2:**
I change gd_config to 'GS'

model_state_file = 'sop_uncropped_resnet50_SG_1536_0.1_0.5_0.1_128_model.pth'
gd_config = 'GS'
num_classes = 11318
state_dict = torch.load(model_state_file, map_location=torch.device('cpu'))
net = Model('resnet50', gd_config, 1536, num_classes)
net.load_state_dict(state_dict)
net.eval()

Loss (Equlidean) = 0.14

For other products - the same.

**Comment by nthhiep** [comment]
Hi, 
I also think this result is not normal.  "sop_uncropped_resnet50_SG_1536_0.1_0.5_0.1_128_model" has 'SG' config.
So, this model must give a smaller loss for 'SG' than for 'GS' config.

Note that 'SG' and 'GS' are two different networks. So, they can make different results for the same input.
